### PR TITLE
test: Code Coverage: My-Profile/ Select Currency #1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 9f85cf4</samp>

This pull request adds and updates unit tests for the `SelectCurrencyComponent`, which lets the user choose a currency. It also adds a mock data object for currencies and imports the necessary dependencies and schemas. The files affected are `src/app/core/mock-data/currency.data.ts` and `src/app/fyle/my-profile/select-currency/select-currency.component.spec.ts`.

### <samp>🤖 Generated by Copilot at 9f85cf4</samp>

> _Mock data object_
> _Testing `SelectCurrency`_
> _Autumn of bugs ends_

### <samp>🤖 Generated by Copilot at 9f85cf4</samp>

*  Add mock data object for currency codes and names ([link](https://github.com/fylein/fyle-mobile-app/pull/2427/files?diff=unified&w=0#diff-4059e027a5753aabf138a09a9afb750f8d93ed403a26ff767d7ba3df2911aaf7R176-R203))
*  Enable and update unit tests for `SelectCurrencyComponent` ([link](https://github.com/fylein/fyle-mobile-app/pull/2427/files?diff=unified&w=0#diff-3322c88547c4e62cff7e390640d165d342e67b54b3e215ab21cd418b0d32ce71L1-R115))

## Clickup
https://app.clickup.com/t/85ztznvhy

## Code Coverage
`95.45% Statements 21/22`
`75% Branches 3/4`
`92.85% Functions 13/14`
`95% Lines 19/20`

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/core/mock-data/currency.data.ts [hunk 1]
```diff
@@ -173,6 +173,34 @@ export const apiAllCurrencies: CurrencyName = {
...
+  BIF: 'Burundian Franc',
+  BMD: 'Bermudan Dollar',
+  BND: 'Brunei Dollar',
+  BOB: 'Bolivian Boliviano',
+  BRL: 'Brazilian Real',
+  BSD: 'Bahamian Dollar',
+  BTC: 'Bitcoin',
+  BTN: 'Bhutanese Ngultrum',
+  BWP: 'Botswanan Pula',
+  BYN: null,
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/core/mock-data/currency.data.ts:201]: Here I was trying to cover a branch, where the data is modelled into `Currency[]`, the value of the key in the object becomes the long name. There is an `OR` operator that assigns the key as the long name of the currency if the value part is missing. If all keys have values, then that branch won't execute

File: src/app/fyle/my-profile/select-currency/select-currency.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,116 @@
...
+    component.ngOnInit();
+    tick(500);
+
+    expect(loaderService.showLoader).toHaveBeenCalledTimes(1);
+    expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+    expect(currencyService.getAll).toHaveBeenCalledTimes(1);
+  }));
+
+  it('ngAfterViewInit(): should update the filteredCurrencies$', fakeAsync(() => {
+    component.searchBarRef = fixture.debugElement.query(By.css('.selection-modal--search-input'));
```
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/my-profile/select-currency/select-currency.component.spec.ts:70]: Can't get access to `nativeElement` here, the methods return `Element` type, but the property requires `ElementRef<HTMLInputElement>`. Type assertions resolves the error, but the tests fail.

**Review by Chethan-Fyle — APPROVED** [review]

**Review by Dimple16 — CHANGES_REQUESTED** [review]
one comment

File: src/app/fyle/my-profile/select-currency/select-currency.component.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jayfyle) [reply src/app/fyle/my-profile/select-currency/select-currency.component.spec.ts:70]: Resolved

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**